packages/cdktf/test/resource.test.ts

```diff
@@ -320,3 +327,193 @@ it("maintains the same order of provisioner", () => {
 
   expect(Testing.synth(stack)).toMatchSnapshot();
 });
+
+it("moves resource to greater nesting in contained construct", () => {
+  const app = Testing.app();
+  const stack = new TerraformStack(app, "test");
+  new TestProvider(stack, "provider", {});
+
+  const construct = new Construct(stack, "construct");
+  const nestedContruct = new Construct(construct, "nested-construct");
+
+  new TestResource(nestedContruct, "simple", {
+    name: "foo",
+    provisioners: [
+      { type: "local-exec", command: "echo 'hello' > world.txt" },
+      { type: "local-exec", command: "echo 'hello' > world1.txt" },
+      { type: "local-exec", command: "echo 'hello' > world2.txt" },
+    ],
+  }).addMoveTarget("test");
+
+  new TestResource(stack, "simple", {
+    name: "foo",
+    provisioners: [
+      { type: "local-exec", command: "echo 'hello' > world.txt" },
+      { type: "local-exec", command: "echo 'hello' > world1.txt" },
+      { type: "local-exec", command: "echo 'hello' > world2.txt" },
+    ],
+  }).moveTo("test");
+
+  const synthedStack = JSON.parse(Testing.synth(stack));
+  expect(synthedStack.moved.from).toEqual("test_resource.simple");
+  expect(synthedStack.moved.to).toEqual(
+    "test_resource.construct_nested-construct_simple_2C3755B0"
+  );
+  expect(Object.keys(synthedStack.resource.test_resource)).toContain(
+    "construct_nested-construct_simple_2C3755B0"
+  );
+  // Must not include old resource being moved from
+  expect(Object.keys(synthedStack.resource.test_resource)).not.toContain(
+    "simple"
+  );
+});
+
+it("moves resource to a lesser nesting from contained construct", () => {
+  const app = Testing.app();
+  const stack = new TerraformStack(app, "test");
+  new TestProvider(stack, "provider", {});
+
+  new TestResource(stack, "simple", {
+    name: "foo",
+    provisioners: [
+      { type: "local-exec", command: "echo 'hello' > world.txt" },
+      { type: "local-exec", command: "echo 'hello' > world1.txt" },
+      { type: "local-exec", command: "echo 'hello' > world2.txt" },
+    ],
+  }).addMoveTarget("test");
+  const construct = new Construct(stack, "construct");
+  const nestedContruct = new Construct(construct, "nested-construct");
+
+  new TestResource(nestedContruct, "simple", {
+    name: "foo",
+    provisioners: [
+      { type: "local-exec", command: "echo 'hello' > world.txt" },
+      { type: "local-exec", command: "echo 'hello' > world1.txt" },
+      { type: "local-exec", command: "echo 'hello' > world2.txt" },
+    ],
+  }).moveTo("test");
```

Comment:
We should also add a test where `moveTo` is called before `addMoveTarget` because I think (as mentioned above) that this might happen in real world applications and if we can make it work nevertheless, that would improve the user experience quite a bit.

But to be fair, this can also be done in a follow-up PR as that's something we can implement "behind the scenes" without changes to the publicly facing APIs.

Reply:
Yea there isn't a test as that currently does not work 😅. Though now that you mention aspects (which I admittedly don't have a great understanding of) I think it'd make sense to include them here to avoid that pain point. I'll give it a go!